homeassistant/components/group/entity.py

```diff
@@ -313,14 +315,17 @@ def _set_tracked(self, entity_ids: Collection[str] | None) -> None:
             registry.state_group_mapping[self.entity_id] = self.single_state_type_key
         else:
             self.single_state_type_key = None
-        self.async_on_remove(self._async_deregister)
+        if not self._async_deregister_hook_set:
+            self.async_on_remove(self._async_deregister)
+            self._async_deregister_hook_set = True
 
         self.trackable = tuple(trackable)
         self.tracking = tuple(tracking)
 
     @callback
     def _async_deregister(self) -> None:
         """Deregister group entity from the registry."""
+        self._async_deregister_hook_set = False
         registry: GroupIntegrationRegistry = self.hass.data[REG_KEY]
```

Comment:
we could store the registry as `self._registry = hass.data[REG_KEY]` so we don't look it up all over this class

Reply:
Could type it as well to avoid having the type hint but probably not worth it for this PR https://developers.home-assistant.io/blog/2024/05/01/improved-hass-data-typing